common/src/View/EntityAttributeEditor.cpp

```diff
@@ -25,42 +25,191 @@
+                document->nodesDidChangeNotifier.removeObserver(this, &EntityAttributeEditor::nodesDidChange);
+            }
+        }
+
+        void EntityAttributeEditor::selectionDidChange(const Selection& selection) {
+            updateIfSelectedEntityDefinitionChanged();
+        }
+
+        void EntityAttributeEditor::nodesDidChange(const Model::NodeList& nodes) {
+            updateIfSelectedEntityDefinitionChanged();
+        }
+
+        void EntityAttributeEditor::updateIfSelectedEntityDefinitionChanged() {
+            MapDocumentSPtr document = lock(m_document);
+            const Assets::EntityDefinition* entityDefinition = Model::AttributableNode::selectEntityDefinition(document->allSelectedAttributableNodes());
+
+            if (entityDefinition != m_currentDefinition) {
+                m_currentDefinition = entityDefinition;
+
+                updateDocumentationAndSmartEditor();
+            }
+        }
+
+        void EntityAttributeEditor::updateDocumentationAndSmartEditor() {
+            MapDocumentSPtr document = lock(m_document);
+            const String& attributeName = m_attributeGrid->selectedRowName();
+
+            m_smartEditorManager->switchEditor(attributeName, document->allSelectedAttributableNodes());
+
+            updateDocumentation(attributeName);
+
+            // collapse the splitter if needed
+            if        (!m_documentationText->IsEmpty() && !m_smartEditorManager->isDefaultEditorActive()) {
+                // show both
+                m_documentationSplitter->restore();
+            } else if ( m_documentationText->IsEmpty() && !m_smartEditorManager->isDefaultEditorActive()) {
+                // show only the smart editor
+                m_documentationSplitter->maximize(m_smartEditorManager);
+            } else if (!m_documentationText->IsEmpty() &&  m_smartEditorManager->isDefaultEditorActive()) {
+                // show only the documentation panel
+                m_documentationSplitter->maximize(m_documentationText);
+            } else {
+                // nothing to display
+                m_documentationSplitter->maximize(m_documentationText);
             }
         }
+
+        void EntityAttributeEditor::updateDocumentation(const String &attributeName) {
+            MapDocumentSPtr document = lock(m_document);
+            const Assets::EntityDefinition* entityDefinition = Model::AttributableNode::selectEntityDefinition(document->allSelectedAttributableNodes());
+
+            m_documentationText->Clear();
+
+            wxTextAttr boldAttr;
+            boldAttr.SetFontWeight(wxFONTWEIGHT_BOLD);
+
+            if (entityDefinition != nullptr) {
+                // add attribute documentation, if available
+                if (const Assets::AttributeDefinition* attributeDefinition = entityDefinition->attributeDefinition(attributeName); attributeDefinition != nullptr) {
```

Comment:
I have become wary of this style. I would prefer to move the variable declaration / assignment into a separate statement.